src/MSBuild/XMake.cs

```diff
@@ -1117,7 +1117,10 @@ bool enableProfiler
                     {
                         try
                         {
-                            if (enableRestore)
+                            // Determine if the user specified /Target:Restore which means we should only execute a restore in the fancy way that /restore is executed
+                            bool restoreOnly = String.Equals(request.TargetNames.SingleOrDefault(), MSBuildConstants.RestoreTargetName, StringComparison.OrdinalIgnoreCase);
```

Comment:
This feels especially magical. Are there any other examples of /t:Foo behaving differently than /t:Bar?

Reply:
This is only for the case when you do `MSBuild /T:Restore` on a CI machine or script where you want to restore.  If the functionality to ignore missing imports only works for `MSBuild.exe /restore`, I thought it would be a bad user experience.  We're already treating `Restore` in such a special way, it seemed like a good idea to do this.